src/platforms/android/androidglean.cpp

```diff
@@ -0,0 +1,133 @@
+  auto vpn = MozillaVPN::instance();
+  connect(vpn, &MozillaVPN::sendGleanPings, this,
+          &AndroidGlean::sendGleanPings);
+  connect(vpn, &MozillaVPN::recordGleanEvent, this,
+          &AndroidGlean::recordGleanEvent);
+  connect(vpn, &MozillaVPN::recordGleanEventWithExtraKeys, this,
+          &AndroidGlean::recordGleanEventWithExtraKeys);
+  connect(vpn, &MozillaVPN::setGleanSourceTags, this,
+          &AndroidGlean::setGleanSourceTags);
+  connect(SettingsHolder::instance(), &SettingsHolder::gleanEnabledChanged,
+          this, &AndroidGlean::gleanEnabledChanged);
+  connect(AndroidVPNActivity::instance(), &AndroidVPNActivity::serviceConnected,
+          this, &AndroidGlean::daemonConnected);
+
+  connect(qApp, &QApplication::applicationStateChanged, this,
+          &AndroidGlean::applicationStateChanged);
+  // Make sure we start the service now, so
+  // glean is ready even before login.
+  AndroidVPNActivity::connectService();
+}
+
+AndroidGlean::~AndroidGlean() {
+  MVPN_COUNT_DTOR(AndroidGlean);
+
+  Q_ASSERT(s_instance == this);
+  s_instance = nullptr;
+}
+
+void AndroidGlean::sendGleanPings() {
+  AndroidVPNActivity::instance()->sendToService(
+      ServiceAction::ACTION_SEND_GLEAN_PING, "");
+}
+
+void AndroidGlean::recordGleanEvent(const QString& gleanSampleName) {
+  QJsonObject args;
+  args["key"] = gleanSampleName;
+  QJsonDocument doc(args);
+  logger.debug() << " recordGleanEvent" << gleanSampleName;
+  AndroidVPNActivity::instance()->sendToService(
+      ServiceAction::ACTION_RECORD_EVENT, doc.toJson(QJsonDocument::Compact));
+}
+
+void AndroidGlean::recordGleanEventWithExtraKeys(const QString& gleanSampleName,
+                                                 const QVariantMap& extraKeys) {
+  QJsonObject extras = QJsonObject::fromVariantMap(extraKeys);
+  QJsonObject args;
+  args["extras"] = extras;
+  args["key"] = gleanSampleName;
+  QJsonDocument doc(args);
+  logger.debug() << " recordGleanEvent" << gleanSampleName;
+  AndroidVPNActivity::instance()->sendToService(
+      ServiceAction::ACTION_RECORD_EVENT, doc.toJson(QJsonDocument::Compact));
+}
+void AndroidGlean::setGleanSourceTags(const QStringList& tags) {
+  // TODO: Current no-op as glean-android does not implement this - no need to
+  // do anything QString list = tags.join(","); auto jList =
+  // QJniObject::fromString(list).object()
+  // QJniObject::callStaticMethod<void>(UTILS_CLASS, "setGleanSourceTag",
+  // "(ZLjava/lang/String;)",jList); return;
+  Q_UNUSED(tags)
+}
+
+void AndroidGlean::gleanEnabledChanged(bool enabled) {
+  QJsonObject args;
+  args["enabled"] = enabled;
+  QJsonDocument doc(args);
+  logger.debug() << " gleanEnabledChanged" << enabled;
+  AndroidVPNActivity::instance()->sendToService(
+      ServiceAction::ACTION_GLEAN_ENABLED_CHANGED,
+      doc.toJson(QJsonDocument::Compact));
+}
+
+void AndroidGlean::daemonConnected() {
+  // Daemon is now ready
+  gleanEnabledChanged(SettingsHolder::instance()->gleanEnabled());
```

Comment:
Does this mean Glean will be started with `uploadEnabled=false` and only when the deamon is connected the upload status will be changed?